Weekly top DEV articles

king-11 opened the pull request:
- nodemon setup
- article module
- lucy-bot endpoint update

nishantwrp commented on .github/workflows/cron.yml:
```diff
@@ -0,0 +1,18 @@
+  ping_bot:
+    runs-on: ubuntu-latest
+
+    name: Lucy Fetch Article on Ping
```
Maybe change it to `Weekly Top DEV Articles`

nishantwrp commented on .env.example:
```diff
@@ -1 +0,0 @@
-LUCY_BOT_TOKEN=
```
Why remove this?

nishantwrp commented on api/article.ts:
```diff
@@ -0,0 +1,44 @@
+import axios from 'axios';
+import { ExtraEditMessage } from 'telegraf/typings/telegram-types';
+import { ContextMessageUpdate, Extra } from 'telegraf';
+interface Article {
```
new line

nishantwrp commented on api/lucy-bot.ts:
```diff
@@ -48,7 +45,18 @@ if (!PROD_ENV) {
+module.exports = async (req: NowRequest, resp: NowResponse): Promise<void> => {
   if (req.method === 'POST') bot.handleUpdate(req.body, resp);
-  else resp.status(200).send('Use POST to use Telegram bot!');
+  else {
+    const replyList = await fetchArticles().then(markdownArticles);
+    await bot.telegram
+      .sendMessage(
+        `${process.env.CHATID}`,
+        `Hey Devs here are week's top blogs :)\n\n${replyList.join('\n\n')}`,
+        <ExtraEditMessage>Extra.markdown().webPreview(false),
+      )
+      .catch((e) => console.error(e));
+
+    resp.status(200).send('Article Sent');
+  }
```
Are changes in this file intentional?

king-11 replied:
@nishantwrp I wasn't able to export the bot properly, the bot has to be there to send a message so I used the request, response of lucy-boy.ts which can be changed to something like a query param check to ensure that no one gets to send a message just on pinging lucy's link although I believe the link is already hidden

nishantwrp replied:
Aren't the articles handled by the `article.ts` file?

nishantwrp commented on nodemon.json:
```diff
@@ -0,0 +1,23 @@
+    "URL": "your",
+    "CHATID": "your"
+  }
+}
```
newline

nishantwrp commented on package.json:
```diff
@@ -11,7 +11,7 @@
   "dependencies": {
     "@octokit/rest": "^18.0.6",
-    "dotenv": "^8.2.0",
+    "axios": "^0.20.0",
```
why remove `dotenv`?

nishantwrp commented on package.json:
```diff
@@ -27,4 +27,4 @@
   },
   "private": true
-}
+}
```
new line

nishantwrp reviewed (changes requested):
Looks good generally. Some comments ptal!

king-11 committed:
update packages and development workflow

nodemon configured to set env variables instead of dotenv
axios added and dotenv removed for dependency
readme.md instructions updated

king-11 committed:
typecheck lucy and better logging

telegraf logging to ensure better error debugging
typecheck on response

king-11 committed:
article.ts module create

fetch article and convert them to markdown syntax
response on passowrd check

king-11 committed:
cron job setup

ping on 08:00 AM every Sunday
remove url from nodemon.json
Update cron.yml

nishantwrp commented on api/lucy-bot.ts:
```diff
@@ -48,7 +44,7 @@ if (!PROD_ENV) {
 
-module.exports = (req: NowRequest, resp: NowResponse) => {
+module.exports = async (req: NowRequest, resp: NowResponse): Promise<NowResponse> => {
   if (req.method === 'POST') bot.handleUpdate(req.body, resp);
-  else resp.status(200).send('Use POST to use Telegram bot!');
+  return resp.status(200).send('Use POST to use Telegram bot!');
```
Is this change required?

king-11 replied:
just additional type checking can be removed

nishantwrp replied:
Okay. then

nishantwrp replied:
Not resolved?

nishantwrp commented on api/lucy-bot.ts:
```diff
@@ -5,13 +5,9 @@ import { ExtraEditMessage } from 'telegraf/typings/telegram-types';
-
 const bot = new Telegraf(process.env.LUCY_BOT_TOKEN || '');
 
+bot.use(Telegraf.log());
```
What does `Telegraph.log()` do exactly?

king-11 replied:
It will provide a detailed log something like this which can be useful to locate issue
```json
{
  "update_id": 24312332,
  "message": {
    "message_id": 173,
    "from": {
      "id": 862301703,
      "is_bot": false,
      "first_name": "Target_X",
      "username": "Target_X",
      "language_code": "en"
    },
    "chat": {
      "id": -450132051,
      "title": "test",
      "type": "group",
      "all_members_are_administrators": true
    },
    "date": 1602702162,
    "text": "/DevTalks",
    "entities": [
      {
        "offset": 0,
        "length": 9,
        "type": "bot_command"
      }
    ]
  }
}
```

nishantwrp replied:
Oh, okay. In vercel logs?

king-11 replied:
yeah

nishantwrp replied:
Cool!

nishantwrp commented on api/article.ts:
```diff
@@ -0,0 +1,67 @@
+  const ms = new Date().getTime() - start.getTime();
+  console.log('Response time: %sms', ms);
+});
+interface Article {
```
add a new line here

king-11 replied:
done

nishantwrp commented on nodemon.json:
```diff
@@ -0,0 +1,24 @@
+  "ext": "js,ts,json",
+  "env": {
+    "LUCY_BOT_TOKEN": "set",
+    "URL": "set",
```
Is this required?

king-11 replied:
URL isn't needed rest are needed for proper working of article endpoint

nishantwrp replied:
Yup, remove it then

king-11 replied:
okay

nishantwrp reviewed (commented):
Few questions, ptal!

nishantwrp commented on .github/workflows/cron.yml:
```diff
@@ -0,0 +1,22 @@
+
+on:
+  schedule:
+    - cron: "11 11 * * 4"
```
oh and this was `* * * * 0` earlier? Is this correct?

king-11 replied:
It was wrong cron syntax "* * * * 0" was every minute on sunday keep on pinging lucy now its for Thursday 11:11

king-11 reviewed (commented):
All changes made

nishantwrp reviewed (approved):
lgtm!
